packages/contracts-bedrock/test/governance/ProposalValidator.t.sol

```diff
@@ -578,11 +656,497 @@ contract ProposalValidator_HashProposalWithModule_Test is ProposalValidator_Init
+    uint128 criteriaValue;
+    string[] optionsDescriptions;
+    address[] optionsRecipients;
+    uint256[] optionsAmounts;
+    string description;
+
+    event ProposalVotingModuleData(bytes32 indexed proposalHash, bytes encodedVotingModuleData);
+
+    function setUp() public override {
+        super.setUp();
+
+        vm.prank(owner);
+        validator.setProposalTypeData(
+            ProposalValidator.ProposalType.GovernanceFund,
+            ProposalValidator.ProposalTypeData({
+                requiredApprovals: PROPOSAL_REQUIRED_APPROVALS,
+                proposalVotingModule: 3
+            })
+        );
+
+        vm.prank(owner);
+        validator.setProposalTypeData(
+            ProposalValidator.ProposalType.CouncilBudget,
+            ProposalValidator.ProposalTypeData({
+                requiredApprovals: PROPOSAL_REQUIRED_APPROVALS,
+                proposalVotingModule: 4
+            })
+        );
+
+        criteriaValue = 1000 ether; // 1000 tokens needed for option to pass
+        optionsDescriptions = new string[](2);
+        optionsDescriptions[0] = "Option A: Fund development";
+        optionsDescriptions[1] = "Option B: Fund marketing";
+
+        optionsRecipients = new address[](2);
+        optionsRecipients[0] = makeAddr("recipient1");
+        optionsRecipients[1] = makeAddr("recipient2");
+
+        optionsAmounts = new uint256[](2);
+        optionsAmounts[0] = 1000 ether;
+        optionsAmounts[1] = 500 ether;
+
+        description = "Test funding proposal";
+    }
+
+    function test_submitFundingProposal_governanceFund_succeeds() public {
+        // Calculate expected proposal hash
+        bytes memory votingModuleData =
+            _constructVotingModuleData(optionsDescriptions, optionsRecipients, optionsAmounts, criteriaValue);
+        bytes32 expectedHash = validator.hashProposalWithModule(
+            makeAddr("approvalVotingModule"), votingModuleData, keccak256(bytes(description))
+        );
+
+        // Mock proposalSnapshot to return 0 (proposal doesn't exist in governor)
+        _mockAndExpect(
+            address(governor),
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        // Expect ProposalSubmitted event
+        vm.expectEmit(address(validator));
+        emit ProposalSubmitted(expectedHash, rando, description, ProposalValidator.ProposalType.GovernanceFund);
+
+        // Expect ProposalVotingModuleData event
+        vm.expectEmit(address(validator));
+        emit ProposalVotingModuleData(expectedHash, votingModuleData);
+
+        vm.prank(rando);
+        bytes32 proposalHash = validator.submitFundingProposal(
+            criteriaValue,
+            optionsDescriptions,
+            optionsRecipients,
+            optionsAmounts,
+            description,
+            ProposalValidator.ProposalType.GovernanceFund
+        );
+
+        assertEq(proposalHash, expectedHash);
```

Comment:
I think this is not enough, we should at least also check that the proposal was stored with the correct data